app/lib/utils/gpDataMapper.js

```diff
@@ -3,4 +3,58 @@ function getBookOnlineLink(gpData) {
     ? gpData.onlineServices.appointments.url : undefined;
 }
 
-module.exports = getBookOnlineLink;
+function getFilteredGps(gpData, searchStr) {
+  if (gpData.doctors) {
+    let filteredSearchStr = searchStr;
+    let filteredDocs = [];
+    if (searchStr.match(new RegExp(/^(doctor|dr.|dr)/, 'i'))) {
+      filteredSearchStr = searchStr.replace(/^\S+ /g, '');
+    }
+    const searchArr = filteredSearchStr.split(' ');
+    searchArr.forEach((searchSubStr) => {
+      const regexp = new RegExp(searchSubStr, 'i');
+      filteredDocs = filteredDocs.concat(gpData.doctors.filter(doctor => regexp.test(doctor)));
+    });
+    return [...new Set(filteredDocs)];
+  }
+  return undefined;
+}
+
+function mappedTitleForGps(doctors) {
+  const mappedTitleGps = [];
+  const regexpDrAll = new RegExp(/^(Other|DOCTOR|Doctor|DR|Dr.|Dt|dr|GP|Dr (Mrs)|Miss|Mr|Mr.|Mra|Mrs|Ms|Senior Partner|Dr)/, 'i');
```

Comment:
Use the literal form of regex creation as opposed to constructor form when the input isn't runtime specific